patches/com/mojang/blaze3d/pipeline/MainTarget.java.patch

```diff
@@ -0,0 +1,51 @@
+--- a/com/mojang/blaze3d/pipeline/MainTarget.java
++++ b/com/mojang/blaze3d/pipeline/MainTarget.java
+@@ -16,8 +_,12 @@
+     static final MainTarget.Dimension DEFAULT_DIMENSIONS = new MainTarget.Dimension(854, 480);
+ 
+     public MainTarget(int p_166137_, int p_166138_) {
+-        super(true);
+-        this.createFrameBuffer(p_166137_, p_166138_);
++        this(net.neoforged.neoforge.client.ClientHooks.configureMainRenderTarget(true, p_166137_, p_166138_));
++    }
++
++    private MainTarget(net.neoforged.neoforge.client.event.ConfigureMainRenderTargetEvent e) {
++        super(e.useDepth(), e.useStencil());
++        this.createFrameBuffer(e.width(), e.height());
+     }
+ 
+     private void createFrameBuffer(int p_166142_, int p_166143_) {
+@@ -37,6 +_,9 @@
+         GlStateManager._texParameter(3553, 10242, 33071);
+         GlStateManager._texParameter(3553, 10243, 33071);
+         GlStateManager._glFramebufferTexture2D(36160, 36096, 3553, this.depthBufferId, 0);
++        if (this.useStencil) {
++            GlStateManager._glFramebufferTexture2D(36160, org.lwjgl.opengl.GL32.GL_STENCIL_ATTACHMENT, 3553, this.depthBufferId, 0);
++        }
+         GlStateManager._bindTexture(0);
+         this.viewWidth = maintarget$dimension.width;
+         this.viewHeight = maintarget$dimension.height;
+@@ -58,8 +_,8 @@
+                 maintarget$attachmentstate = maintarget$attachmentstate.with(MainTarget.AttachmentState.COLOR);
+             }
+ 
+-            if (this.allocateDepthAttachment(maintarget$dimension)) {
+-                maintarget$attachmentstate = maintarget$attachmentstate.with(MainTarget.AttachmentState.DEPTH);
++            if (this.useStencil && this.allocateDepthAttachment(maintarget$dimension)) {
++                maintarget$attachmentstate = maintarget$attachmentstate.with(AttachmentState.DEPTH);
+             }
+ 
+             if (maintarget$attachmentstate == MainTarget.AttachmentState.COLOR_DEPTH) {
+@@ -82,7 +_,11 @@
+         RenderSystem.assertOnRenderThreadOrInit();
+         GlStateManager._getError();
+         GlStateManager._bindTexture(this.depthBufferId);
+-        GlStateManager._texImage2D(3553, 0, 6402, p_166145_.width, p_166145_.height, 0, 6402, 5126, null);
++        if (this.useStencil) {
++            GlStateManager._texImage2D(3553, 0, 6402, p_166145_.width, p_166145_.height, 0, org.lwjgl.opengl.GL32.GL_DEPTH_STENCIL, org.lwjgl.opengl.GL32.GL_UNSIGNED_INT_24_8, null);
```

Comment:
I believe the original code used `GL_UNSIGNED_INT_24_8_REV`. We should also look into the parameter again, Jelly said on discord that vanilla uses 32 bits for depth so we should preserve that.